src/utils/hooks/useOutcomes.tsx

```diff
@@ -0,0 +1,122 @@
+  // State (perhaps it should be?)
+  const [error, setError] = useState<ErrorResponse | null>(null)
+  const [loading, setLoading] = useState(false)
+  const [outcomesList, setOutcomesList] = useState<OutcomesResponse>([])
+
+  useEffect(() => {
+    const JWT = localStorage.getItem(LOCAL_STORAGE_JWT_KEY)
+    if (!JWT) return
+    const url = `${serverURL}/outcomes`
+    processRequest(url, JWT, setError, setLoading, setOutcomesList)
+  }, [templatePermissions])
+
+  return { error, loading, outcomesList }
+}
+
+export const useOutcomesTable = ({
+  templatePermissions,
+  tableName,
+  apiQueries,
+}: OutcomeTableProps) => {
+  const { first, offset, orderBy, ascending } = apiQueries
+  const [error, setError] = useState<ErrorResponse | null>(null)
+  const [loading, setLoading] = useState(false)
+  const [outcomeTable, setOutcomeTable] = useState<OutcomesTableResponse>()
+
+  useEffect(() => {
+    const JWT = localStorage.getItem(LOCAL_STORAGE_JWT_KEY)
+    if (!JWT) return
+    const queryElements = []
+    if (first) queryElements.push(`first=${first}`)
+    if (offset) queryElements.push(`offset=${offset}`)
+    if (orderBy) queryElements.push(`orderBy=${orderBy}`)
+    if (ascending) queryElements.push(`ascending=${ascending}`)
+    const queryString = queryElements.join('&')
+    const url = `${serverURL}/outcomes/table/${tableName}?${queryString}`
+    processRequest(url, JWT, setError, setLoading, setOutcomeTable)
+  }, [templatePermissions, tableName, apiQueries])
+
+  return { error, loading, outcomeTable }
+}
+
+export const useOutcomesDetail = ({
+  templatePermissions,
+  tableName,
+  recordId,
+}: OutcomeDetailsProps) => {
+  const [error, setError] = useState<ErrorResponse | null>(null)
+  const [loading, setLoading] = useState(false)
+  const [outcomeDetail, setOutcomeDetail] = useState<OutcomesDetailResponse>()
+
+  useEffect(() => {
+    const JWT = localStorage.getItem(LOCAL_STORAGE_JWT_KEY)
+    if (!JWT) return
+    const url = `${serverURL}/outcomes/table/${tableName}/item/${recordId}`
+    processRequest(url, JWT, setError, setLoading, setOutcomeDetail)
+  }, [templatePermissions])
+
+  return { error, loading, outcomeDetail }
+}
+
+const processRequest = (
+  url: string,
+  JWT: string,
+  setErrorMethod: (_: ErrorResponse | null) => void,
+  setLoadingMethod: (_: boolean) => void,
+  setStateMethod: (_: any) => void
+): void => {
+  const setState = (errorState: ErrorResponse | null, loadingState: boolean, stateState: any) => {
+    setErrorMethod(errorState)
+    setLoadingMethod(loadingState)
+    setStateMethod(stateState)
+  }
+  setLoadingMethod(true)
+  getRequest(url, { Authorization: `Bearer ${JWT}` })
+    .then((response) => {
+      if (response?.statusCode) {
+        setState(response, false, undefined)
+        return
+      } else setState(null, false, response)
+    })
+    .catch((error) => {
+      setState(error, false, undefined)
+    })
+}
```

Comment:
I checked that different user can still get access to many outcomes that shouldn't be accessible to them, such as other organizations applied by other users and other users in the system. Was wondering if we should add the restrictions as part of this (or maybe in a part-2 PR) to start adding view restrictions to applicant users to only be able to view their outcomes and review users to be able to see all outcomes (for now)

Reply:
The Outcomes API handles column-based restrictions, not row-based, so the reason you're currently seeing what you're seeing is because we don't have these policies applied to user and org table yet. The mechanism is in here -- it passes the user's JWT to the GraphQL query, but the table is not yet restricted.

There is an issue (https://github.com/openmsupply/application-manager-server/issues/589) for Andrei to address this. However, we also realised that (for outcomes that aren't "standard" i.e. user and org) we should implement some row-based restrictions in Outcomes to handle these, since dynamically adding policies to new tables is tricky -- hence [this issue](https://github.com/openmsupply/application-manager-server/issues/590) to deal with that.

Reply:
Great. Thanks for having already discussed this with Andrei. I'll have a look on these issues now